.github/workflows/test-docker-build.yml

```diff
@@ -1,42 +1,61 @@
-
-concurrency:
-  group: docker-image-builds
-  cancel-in-progress: false
+    paths:
+      # Run only when DockerFile files are modified
+      - "docker/**"
 
 jobs:
-  name: "Build all modified docker images"
-  runs-on: ubuntu-latest
-  steps:
-    - name: Set up Docker Buildx
-      uses: docker/setup-buildx-action@v1
-    - name: Check out code
-      uses: actions/checkout@v3
-    - name: Get changed files
-      id: changed-files
-      uses: tj-actions/changed-files@3f54ebb830831fc121d3263c1857cfbdc310cdb9 #v42
-      with:
-        files: docker/**
-    - name: Run step if only the files listed above change
-      if: steps.changed-files.outputs.only_changed == 'true'
-      env:
-        CHANGED_FILES: ${{ steps.changed-files.outputs.all_changed_files}}
-      run: |
-        for file in ${CHANGED_FILES}; do
-          echo "$file was changed"
-        done
-    - name: Build Docker images
-      strategy:
-        matrix:
-          docker-file: ${{ steps.changed-files.outputs.all_changed_files}}
-      uses: docker/build-push-action@v4
-      with:
-        context: ${{ matrix.docker-file }}
-        push: False
-
+  get_changed_files:
+    name: "Build all modified docker images"
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@3f54ebb830831fc121d3263c1857cfbdc310cdb9 #v42
+        with:
+          files: docker/**
+          json: "true"
+      - name: Run step if only the files listed above change
+        if: steps.changed-files.outputs.any_changed == 'true'
+        id: set-matrix
+        run: |
+          for file in ${{ steps.changed-files.outputs.all_changed_files}}; do
+            echo "$file was changed"
+          done 
+          echo "matrix=${{ steps.changed-files.outputs.all_changed_files}}" >> $GITHUB_OUTPUT
+  build_modified_files:
+    needs: get_changed_files
+    name: Build Docker images on modified files
+    runs-on: ubuntu-latest
+    if: ${{ needs.get_changed_files.outputs.matrix }} != ''
+    strategy:
+      fail-fast: false
+      matrix:
+        docker-file: ${{ fromJson(needs.get_changed_files.outputs.matrix) }}
+    steps:
+      - name: Cleanup disk
+        run: |
+          sudo ls -l /usr/local/lib/
+          sudo ls -l /usr/share/
+          sudo du -sh /usr/local/lib/
+          sudo du -sh /usr/share/
+          sudo rm -rf /usr/local/lib/android
+          sudo rm -rf /usr/share/dotnet
+          sudo du -sh /usr/local/lib/
+          sudo du -sh /usr/share/
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Build Docker image
+        uses: docker/build-push-action@v4
+        with:
+          file: ${{ matrix.docker-file }}
+          context: .
```

Comment:
I am not so familiar, but probably we don't need `context` here if `file` is specified